### PR TITLE
ddtrace/tracer: add WithServiceNameMatch option

### PR DESCRIPTION
**Pull request by dianashevchenko**
Removed limitation of having service name match the name defined when starting the tracer for version tracking.

Closes #1161

### CODE REVIEW

File: ddtrace/tracer/option.go [hunk 1]
```diff
@@ -532,6 +537,14 @@ func WithServiceName(name string) StartOption {
 	}
 }
 
+// WithServiceNameMatch allows specifying whether span service name and config service name
+// should match to set application version tag
```
Comment (knusbaum) [inline ddtrace/tracer/option.go:541]:
```suggestion
// WithServiceNameMatch allows specifying whether span service name and config service name
// must match for the tracer to set the application version tag.
// See: WithService, WithServiceVersion
```

**Review by knusbaum — APPROVED** [review]
Looks great.

**Comment by xrn** [comment]
I do not get this PR - title is saying that work is on WithServiceNameMatch and then code is changing WithUniversalVersion - can someone explain?

**Comment by dianashevchenko** [comment]
@xrn sorry for confusion, I updated the PR name. Since the change concerned setting version tag on the span, it was resolved to rename the option to `WithUniversalVersion`, which makes more sense. With this option, version span tag will be set on all the spans. Hope that helps 🙌

**Pull request merged**